portal/src/main/java/com/liferay/faces/portal/component/inputsearch/internal/InputSearchRenderer.java

```diff
@@ -241,7 +239,10 @@ protected void dynamicallyAddChildComponents(InputSearch inputSearch) {
 			if (!showButton && (hasButtonAjaxBehavior || (action != null) || (actionListener != null))) {
 				logger.error("Set showButton=\"true\" when using action/actionListener or f:ajax.");
 			}
+
+			facesContext.getViewRoot().markInitialState();
```

Comment:
This is the one! I forgot that in JSF 2 there is a partial state. To maintain this properly, if some dynamic attributes are added it should be told to ViewRoot so it can store the new state and restore in next postbacks.